quickwit/quickwit-proto/protos/quickwit/search.proto

```diff
@@ -111,6 +115,57 @@ message ReportSplitsRequest {
 
 message ReportSplitsResponse {}
 
+// -- ListFields -------------------
+
+message ListFieldsRequest{
+  // Optional limit query to a set of indexes.
+  repeated string index_id = 1;
+  // Optional limit query to a set of splits.
+  repeated string split_id = 2;
+  // Optional limit query to a list of fields
+  // Wildcard expressions are supported.
+  repeated string fields = 3;
+
+  // Control if the the request will fail if split_ids contains a split that does not exist. 
+  // optional bool fail_on_missing_index = 3;
+}
+
+message LeafListFieldsRequest{
+  // The index id
+  string index_id = 1;
+  // The index uri
+  string index_uri = 2;
+  // Index split ids to apply the query on.
+  // This ids are resolved from the index_uri defined in the search_request.
+  repeated SplitIdAndFooterOffsets split_offsets = 3;
+
+  // Optional limit query to a list of fields
+  // Wildcard expressions are supported.
+  repeated string fields = 4;
+}
+
+message ListFieldsResponse{
+  repeated ListFieldsEntryResponse fields = 1;
+}
+
+message ListFieldsEntryResponse{
+  // The field name
+  string field_name = 1;
+  // The tantivy field type
+  int32 field_type = 2;
```

Comment:
Can you comment on how it is mapped? Why signed int?
 
```suggestion
  int32 field_type = 2;
```

Reply:
I added a comment (`Type::to_code()`) and switched to u32